buf/registry/owner/v1/user_service.proto

```diff
@@ -97,12 +97,14 @@ message ListUsersRequest {
   // Only return Users of these types.
   repeated UserType has_types = 5 [
     (buf.validate.field).repeated.unique = true,
-    (buf.validate.field).repeated.items.enum.defined_only = true
+    (buf.validate.field).repeated.items.enum.defined_only = true,
+    (buf.validate.field).repeated.items.enum.not_in = 0
   ];
   // Only return Users of these states.
   repeated UserState has_states = 6 [
     (buf.validate.field).repeated.unique = true,
-    (buf.validate.field).repeated.items.enum.defined_only = true
+    (buf.validate.field).repeated.items.enum.defined_only = true,
+    (buf.validate.field).repeated.items.enum.not_in = 0
```

Comment:
Makes sense to add unique to these as well.

Reply:
I avoided adding unique constraints as it was unclear what hit it'd have on perf as n approaches infinity - perhaps not a practical concern, but I'm not sure about them

Reply:
The unique check is O(n) with fail-fast behavior (at least in the go impl). Basically, shoves the values into a `map[T]struct{}` which does mean there's some allocation cost. Unless you've got a massive amount of values, though it's fairly negligible.